kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MetadataDoc.java

```diff
@@ -18,6 +18,7 @@
 
 import java.util.Objects;
 
+// TODO (sophie): rename to KVMetadataDoc and eliminate redundant partition field
```

Comment:
I kind-of liked having `partition` be the name of the partition field, since `_id` is pretty opaque. I think there's a way to configure a table not to use `_id` and use a custom field instead, but it's hardly any extra data overhead I figured the duplication is fine

Reply:
Seems silly to track extra data, although it's only an extra 4 bytes per kafka partition per state store, so the overall size isn't too bad

I'll look into whether we can use a custom field name instead of `_id` but if not, I guess we can just track the partition separately as well